### PR TITLE
test(movie struct): Added basic unit tests for the movie struct and the database

### PR DESCRIPTION
**Pull request by peterdolan**
What: Adding test coverage to the 'movie' in the database.
Why: To get test coverage!

Based off [#1](https://github.com/peterdolan/sfmovies/pull/1), [diff](https://github.com/peterdolan/sfmovies/compare/base...add_tests)

### CODE REVIEW

**Comment by charlesbjohnson** [comment]
@peterdolan can you add a small description including what & why to this? Also, you should change the PR title to match the last commit.

**Comment by charlesbjohnson** [comment]
One other thing. It's pretty common to branch off of a branch that's already up for code review. In these cases, the diff in your latest PR ends up looking like it includes all the changes from the previous PR. To help make life easier for the code reviewers, it helps to include a link that highlights the actual changes in the latest PR (ie. the changes between branches). [Here's a PR](https://github.com/lob/infrastructure/pull/76) I put up the other day where you can see how I included the real diff at the top of the description.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+language: node_js
+sudo: false
+node_js:
+  - 5
+env:
+  - NODE_ENV="test"
+before_script:
+  - npm run db:setup:user
+  - npm run db:resetj
```
Comment (charlesbjohnson) [inline .travis.yml]: looks like a `j` snuck in

File: db/seeds/data/movies.json [hunk 2]
```diff
@@ -0,0 +1,1096 @@
+[{  
+  "title": "180",
+  "release_year": 2011
+},
+{
+  "title": "24 Hours on Craigslist",
```
Comment (charlesbjohnson) [inline db/seeds/data/movies.json]: Looking for bicycles.

File: db/seeds/data/movies.json [hunk 3]
```diff
@@ -0,0 +1,1096 @@
...
+{
+  "title": "Hello Frisco, Hello",
+  "release_year": 1943
+},
+{
+  "title": "Hemingway & Gelhorn",
+  "release_year": 2011
+},
+{
+  "title": "Herbie Rides Again",
```
Comment (charlesbjohnson) [inline db/seeds/data/movies.json]: Wow, I had no idea the Lindsay Lohan version was a remake.

File: lib/index.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+'use strict';
+const Util = require('util');
```
Comment (charlesbjohnson) [inline lib/index.js]: nit: newline after `use strict`

File: test/models/movie.test.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+const Movie = require('../../lib/models/movie');
+
+describe('movie model', () => {
```
Comment (charlesbjohnson) [inline test/models/movie.test.js]: nit: newline after `describe`s

File: test/models/movie.test.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    it('includes all of the necessary fields', () => {
+      const movie = Movie.forge().serialize();
+      expect(movie).to.have.all.keys([
+        'id',
+        'title',
+        'release_year',
+        'object'
+      ]);
+    });
+  });
```
Comment (charlesbjohnson) [inline test/models/movie.test.js]: nit: newline after `describe` end

Comment (charlesbjohnson) [inline test/models/movie.test.js]: Here's [an example](https://github.com/lob/lob-api/blob/master/test/models/statement.test.js) of an actual test in lob-api if you wanted to get a more holistic sense of style/formatting

**Comment by charlesbjohnson** [comment]
Done with a first pass, you can re-assign me when you're ready for me to take another look 👍

File: test/models/movie.test.js (hunk 5 above)
Comment (peterdolan) [inline test/models/movie.test.js]: Ahh okay.

File: lib/models/movie.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+const Bookshelf = require('../library/bookshelf');
```
Comment (charlesbjohnson) [inline lib/models/movie.js]: optional: `library` is fine, but for context we [usually call](https://github.com/lob/signal/tree/master/lib/libraries) this directory `libraries`

File: lib/server.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  }
+});
+
+server.connection({ port: Config.PORT});
+
+server.register([
+  require('hapi-bookshelf-serializer')
+], (err) => {
+  //istanbul ignores
+  if(err){
```
Comment (charlesbjohnson) [inline lib/server.js]: nit: space after `if`, and before `{`

**Comment by charlesbjohnson** [comment]
LGTM, but let's make sure we get #1 merged first. After that we can rebase this off of `master` and check for any other lint issues.

**Comment by charlesbjohnson** [comment]
2 things:

- [ ] check travis for all the places where the lint complained
- [ ] it looks like the tests in travis are also failing due to some database issues. you'll need to update your `.travis.yml` to setup the database tables and whatnot before it begins a test run. [Relevant miyagi](https://github.com/lob/miyagi/blob/master/06-build-your-own-api/07-migrations.md#2-run-migrations-in-the-build) section.

**Comment by peterdolan** [comment]
All good!

**Comment by charlesbjohnson** [comment]
Nice! LGTM

**Pull request merged**